tests/Test.Yaapii.Xambly/Directive/NsDirectiveTest.cs

```diff
@@ -48,40 +49,262 @@ public void SetsNsAttr()
-            //        dom,
-            //        new DomCursor(
-            //            new ManyOf<XNode>(root)
-            //        ),
-            //        new DomStack()
-            //);
+        [Fact]
+        public void StroresPrefixedNamespaceInRoot()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("child")
+                    .Ns("MyNiceNamespace", "my")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<root xmlns:my=\"MyNiceNamespace\"><my:child /></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void ReplacesPrefixedNamespace()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("child")
+                    .Ns("MyOldUglyNamespace", "my")
+                    .Ns("MyNewNiceNamespace", "my")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<root xmlns:my=\"MyNewNiceNamespace\"><my:child /></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void ChildsInheritsNamespace()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("parent")
+                        .Add("child")
+                .Xpath("/root")
+                .Ns("MyNiceNamespace", "nice")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<nice:root xmlns:nice=\"MyNiceNamespace\"><nice:parent><nice:child /></nice:parent></nice:root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void AddsPrefixedNamespaceToSeveralNodes()
+        {
+            var dom = new XDocument();
+            new Xambler(
+                new Directives()
+                .Add("root")
+                    .Add("parent")
+                        .Add("child")
+                .Xpath("/root")
+                    .Add("anotherParent")
+                        .Add("anotherChild")
+                .Xpath("/root/*")
+                .Ns("MyNiceNamespace", "nice")
+            ).Apply(dom);
+
+            Assert.Equal(
+                "<root xmlns:nice=\"MyNiceNamespace\"><nice:parent><nice:child /></nice:parent><nice:anotherParent><nice:anotherChild /></nice:anotherParent></root>",
+                dom.ToString(SaveOptions.DisableFormatting)
+            );
+        }
+
+        [Fact]
+        public void PrefixedNamespaceAffectesAttributes()
+        {
+            var dom = new XDocument();
+            var nsResolver = new XmlNamespaceManager(new NameTable());
+            nsResolver.AddNamespace("nice", "MyNiceNamespace");
```

Comment:
is the resolver necessary for this test?

Reply:
removed after refactoring